src/game/client/worldheightmap.cpp

```diff
@@ -29,21 +45,2359 @@ class GDIFileStream : public InputStream
+{
+    File *file;
+    TerrainType *terrain = g_theTerrainTypes->Find_Terrain(tex_class->name);
+
+    if (terrain) {
+        char fname[260];
+        sprintf(fname, "%s%s", "Art/Terrain/", terrain->Get_Texture().Str());
+        file = g_theFileSystem->Open(fname, File::BINARY | File::READ);
+    } else {
+        file = g_theFileSystem->Open(tex_class->name, File::BINARY | File::READ);
+    }
+
+    if (file) {
+        GDIFileStream stream(file);
+        int numTiles = Count_Tiles(&stream, nullptr);
+        file->Seek(0, File::START);
+
+        if (numTiles >= tex_class->num_tiles) {
+            numTiles = tex_class->num_tiles;
+            int i;
+
+            for (i = 10; i >= 1; i--) {
+                if (numTiles >= i * i) {
+                    numTiles = i * i;
+                    break;
+                }
+            }
+
+            Read_Tiles(&stream, &tile_data[tex_class->first_tile], i);
+        }
+
+        file->Close();
+    }
+}
+
+bool WorldHeightMap::Parse_Object_Data_Chunk(DataChunkInput &file, DataChunkInfo *info, void *user_data)
+{
+    return ((WorldHeightMap *)file.m_userData)->Parse_Object_Data(file, info, user_data, info->version >= 2);
+}
+
+bool WorldHeightMap::Parse_Object_Data(DataChunkInput &file, DataChunkInfo *info, void *user_data, bool read_dict)
+{
+    MapObject *pPrevious = (MapObject *)file.m_currentObject;
+    Coord3D loc;
+    loc.x = file.Read_Real32();
+    loc.y = file.Read_Real32();
+    loc.z = file.Read_Real32();
+    float f1 = -1000.0f;
+    float f2 = 1593.75f;
+
+    if (info->version <= 2) {
+        loc.z = 0.0f;
+    }
+
+    float angle = file.Read_Real32();
+    int flags = file.Read_Int32();
+    Utf8String name = file.Read_AsciiString();
+    Dict d(0);
+
+    if (read_dict) {
+        d = file.Read_Dict();
+    }
+
+    if (loc.z >= f1 && loc.z <= f2) {
+        MapObject *obj = NEW_POOL_OBJ(MapObject, loc, name, angle, flags, &d, g_theThingFactory->Find_Template(name, false));
+
+        if (obj->Get_Properties()->Get_Type(g_waypointIDKey) == Dict::DICT_INT) {
+            obj->Set_Is_Waypoint();
+        }
+
+        if (obj->Get_Properties()->Get_Type(g_lightHeightAboveTerrainKey) == Dict::DICT_REAL) {
+            obj->Set_Is_Light();
+        }
+
+        if (obj->Get_Properties()->Get_Type(g_scorchTypeKey) == Dict::DICT_INT) {
+            obj->Set_Is_Scorch();
+        }
+
+        if (pPrevious) {
+            captainslog_dbgassert(MapObject::s_theMapObjectListPtr && !pPrevious->Get_Next(), "Bad linkage.");
+
+            pPrevious->Set_Next_Map(obj);
+        } else {
+            captainslog_dbgassert(!MapObject::s_theMapObjectListPtr, "Bad linkage.");
+            MapObject::s_theMapObjectListPtr = obj;
+        }
+
+        file.m_currentObject = obj;
+    } else {
+        captainslog_debug("Removing object at z height %f", loc.z);
+    }
+
+    return true;
+}
+
+struct TTargaHeader

```

Comment:
use TGAHeader, lets not add a redundant copy

Reply:
fixed.